pvlib/test/test_ivtools.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May  9 10:51:15 2019
+
+@author: cwhanse
+"""
+
+import numpy as np
+import pandas as pd
+import pytest
+from pvlib import pvsystem
+from pvlib import ivtools
+from conftest import requires_scipy, requires_pysam
+
+
+def get_test_iv_params():
+    return {'IL': 8.0, 'I0': 5e-10, 'Rsh': 1000, 'Rs': 0.2, 'nNsVth': 1.61864}
+
+
+def get_cec_params_cansol_cs5p_220p():
+    return {'V_mp_ref': 46.6, 'I_mp_ref': 4.73, 'V_oc_ref': 58.3,
+            'I_sc_ref': 5.05, 'alpha_isc': 0.000495, 'beta_voc': -0.003372,
+            'gamma_pmp': -0.43, 'cells_in_series': 96}
+
```

Comment:
Just curious, why are these functions?

Reply:
To be re-used in tests for other fitting methods.

Reply:
Still curious to understand what flexibility the function adds compared to a dict with the same name...

Reply:
What's nice about the function is that it returns a copy of the dictionary that can be mutated without mutating the original dictionary. In testing, this can be useful for changing one or two parameters off of a base case.

Reply:
Ok, instead of `dict.copy()`?  The luxury of always having multiple ways to achieve something in python!  :)

Reply:
Inadvertently mutating dictionaries (or other objects) in Python remains one of my biggest challenges with the language. I admittedly wasn't sure of the behavior here until I ran some examples.

Reply:
When using `pytest`, it's preferable mark the setup function with `@pytest.fixture` and then pass the name of the setup function into the test function. `pytest` calls the fixture each time a new test is run (unless otherwise specified using the `scope` keyword argument). So this function becomes:

```
@pytest.fixture
def cec_params_cansol_cs5p_220p():
    return {'V_mp_ref': 46.6, 'I_mp_ref': 4.73, 'V_oc_ref': 58.3,
            'I_sc_ref': 5.05, 'alpha_sc': 0.0025, 'beta_voc': -0.19659,
            'gamma_pmp': -0.43, 'cells_in_series': 96}
```

and the test function becomes

```
def test_fit_cec_sam(cec_module_parameters, cec_params_cansol_cs5p_220p):
# no need to call get_cec_params_cansol_cs5p_220p()
```

Reply:
what is the returned dict assigned to, so it can be referenced?

Reply:
When `pytest` runs the test function, the fixture's returned dict is assigned to `cec_params_cansol_cs5p_220p` within the test function. `pytest` creates new returned dict if the fixture is used in another test function. https://docs.pytest.org/en/latest/fixture.html